### PR TITLE
[BEAM-1345] Clearly delineate public API in apache_beam/options

### PR DESCRIPTION
**Pull request by charlesccychen**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by charlesccychen** [comment]
R: @robertwb 
CC: @aaltay, @mariapython

File: sdks/python/apache_beam/options/pipeline_options.py [hunk 1]
```diff
@@ -25,6 +25,21 @@
...
+    'StandardOptions',
+    'TypeOptions',
+    'DirectOptions',
+    'GoogleCloudOptions',
+    'WorkerOptions',
+    'DebugOptions',
+    'ProfilingOptions',
+    'SetupOptions',
+    'TestOptions',
+    'OptionsContext',
```
Comment (aaltay) [inline sdks/python/apache_beam/options/pipeline_options.py]: Should move `OptionsContext` out of the public api?

Comment (charlesccychen) [inline sdks/python/apache_beam/options/pipeline_options.py]: Done.

**Comment by aaltay** [comment]
LGTM

**Pull request closed without merging**